search/models.py

```diff
@@ -68,10 +68,13 @@ def serve(self, request, page=None):
             search_results = sorted(search_results, key=lambda x: x._score, reverse=True)
 
             promoted = [x.page.specific for x in Query.get(search_query).editors_picks.all() if x.page.live]
+            promoted_page_ids = [promoted_page.id for promoted_page in promoted]
+            filtered_search_results = [result for result in search_results if result.id not in promoted_page_ids]
```

Comment:
Wondering if we can't run this logic in the queryset? I am not so good with Wagtail, but should be technically possible as I know it's possible in Django. @akmiller01 your thoughts?

Reply:
I think it would be more straight forward if `search_results` was a queryset instead of a list. Then you could do `search_results.exclude(id__in=promoted_page_ids)`. But since search_results is a list of multiple model querysets, it won't be that easy. Would need to combine list into a queryset with the `Q` function. Which is possible, but Thatcher's solution here is readable which is nice.